## Inference

Setelah menyimpan model dalam bentuk file, kita bisa memanggilnya kembali untuk menguji bagaimana model melakukan prediksi. Ini sangat membantu untuk mengevaluasi seberapa baik model bekerja dan bagaimana kemampuannya memprediksi data baru. Dengan cara ini, kita bisa menilai seberapa akurat model dalam menghadapi situasi dunia nyata. Proses ini menjadi kunci untuk memastikan dan menguji model GRU yang sudah kita latih sebelumnya.

In [2]:
import pandas as pd
from keras.models import load_model
import numpy as np

# Load Model
model = load_model('model_gru_2')

# Input Text
input_text = "you are amazing"

# Make a prediction using model
predictions = model.predict(np.array([input_text]))

predicted_class = np.argmax(predictions[0])

# Map predicted class to labels
class_labels = {0: 'Negative', 1: 'Positive', 2: 'Neutral'}
predicted_label = class_labels[predicted_class]

# Display the results
print("Input Text:", input_text)
print("Predicted Class:", predicted_class)
print("Predicted Label:", predicted_label)
print("Prediction Probabilities:", predictions[0])

# Create a DataFrame to store the results
columns = ['Input Text', 'Predicted Class', 'Predicted Label', 'Prediction Probabilities']
data = [[input_text, predicted_class, predicted_label, predictions[0]]]
result_df = pd.DataFrame(data, columns=columns)

# Display the DataFrame
print("\nResult DataFrame:")
print(result_df)


1/1 [==============================] - 2s 2s/step
Input Text: you are amazing
Predicted Class: 1
Predicted Label: Positive
Prediction Probabilities: [0.18456388 0.5119029  0.3035332 ]

Result DataFrame:
        Input Text  Predicted Class Predicted Label  \
0  you are amazing                1        Positive   

             Prediction Probabilities  
0  [0.18456388, 0.5119029, 0.3035332]  


**insight** :

1. Proses inferensi data bertujuan untuk menguji sejauh mana mengklasifikasikan sentimen dari teks yang diberikan, serta sejauh mana model yakin dengan prediksi yang dihasilkan menggunakan dummy data.
2. pada data inference. dummy Teks yang digunakan sebagai input untuk model adalah "`you are amazing`", Kelas yang diprediksi oleh model adalah kelas dengan nilai 1, Label yang sesuai dengan kelas yang diprediksi adalah "Positive". Dalam konteks ini, kelas 1 diinterpretasikan sebagai sentimen positif. jika dilihat dari Probabilitas prediksi untuk masing-masing kelas. Dalam hal ini, model memberikan probabilitas sekitar 18.46% untuk kelas 0 (Negatif), 51.19% untuk kelas 1 (Positif), dan 30.35% untuk kelas 2 (Netral). maka dari itu model di prediksi sebagai class 1 atau Positive